### PR TITLE
neverest: fix darwin build

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

WIP

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by TomaSajt** [comment]
@ofborg build neverest

File: pkgs/by-name/ne/neverest/package.nix [hunk 1]
```diff
@@ -34,7 +34,43 @@ rustPlatform.buildRustPackage rec {
     ++ lib.optional (installManPages || installShellCompletions) installShellFiles;
 
   buildInputs = [ ]
-    ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [ AppKit Cocoa Security ])
+  ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk_11_0.frameworks; [
```
Comment (soywod) [inline pkgs/by-name/ne/neverest/package.nix:37]: What is the impact of selecting apple sdk 11 versus the default one?

File: pkgs/by-name/ne/neverest/package.nix [hunk 2]
```diff
@@ -34,7 +34,43 @@ rustPlatform.buildRustPackage rec {
...
+  ImageIO
+  Metal
+  NetFS
+  OpenCL
+  OpenGL
+  QuartzCore
+  Security
+  ServiceManagement
+  SystemConfiguration
+  ])
```
Comment (soywod) [inline pkgs/by-name/ne/neverest/package.nix:73]: How come do we need all those frameworks? Is there any other Nix package with such definition?

**Review by soywod — COMMENTED** [review]
Since I don't have a Mac I cannot test those changes. Is it a good practice to add so many frameworks to the build input?

File: pkgs/by-name/ne/neverest/package.nix (hunk 2 above)
Comment (TomaSajt) [reply pkgs/by-name/ne/neverest/package.nix:73]: I don't have a Mac either, I'm just trying to work off of the hydra/CI logs and see if the CI passes (it does not, not sure why). If I can't get it to work I'll close this.

**Comment by TomaSajt** [comment]
@ofborg build neverest

**Comment by soywod** [comment]
How can I access log history? I cannot find the initial error you got. I would like to help you.

**Comment by TomaSajt** [comment]
> How can I access log history? I cannot find the initial error you got. I would like to help you.

I was looking at random package faliures from https://zh.fail/failed/all.html
And found `neverest`'s failure: https://hydra.nixos.org/build/259369057
I saw that there were some symbol-not-found errors, which usually mean that a later MacOS/Apple SDK version is required.
So I tried `apple_sdk_11_0` and opened this PR to see if OfBorg could build it. The many frameworks you were seeing earlier were probably uneeded, just wanted to check something.

I now think that not only does this need `apple_sdk_11_0` for the frameworks, but also for `stdenv` itself, which is why I've added `darwin.apple_sdk_11_0.callPackage`. Unfortunately, OfBorg is kind of slow with darwin builds, so I'm still waiting for the CI to see if it helped or not.

Though, if this fixes it then we'll need to do something about the fact that `pkgs/by-name` can't use anything other than the normal `callPackage`

**Comment by soywod** [comment]
> And found `neverest`'s failure:
>
> <https://hydra.nixos.org/build/259369057>

I got a similar error with another project of mine. I needed to add
Cocoa framework to the buildInput and manually set up NIX_LDFLAGS:

<https://github.com/soywod/himalaya/blob/7a951b48303c76e6fad18c63f9105c3c3ac94d12/flake.nix#L79-L80>

**Comment by TomaSajt** [comment]
Just wanted to say that I will not be available for 3-4 days, so if you want to fix the package, you'll probably want to create a separate PR.

**Comment by soywod** [comment]
> Just wanted to say that I will not be available for 3-4 days, so if
> you want to fix the package, you'll probably want to create a separate
> PR.

Alright, I may try then. Thank you for initiating this!

**Pull request closed without merging**